src/Microsoft.Extensions.OData.Migration/ExtensionMethods.cs

```diff
@@ -0,0 +1,143 @@
+                case EdmPrimitiveTypeKind.Date:
+                case EdmPrimitiveTypeKind.Double:
+                case EdmPrimitiveTypeKind.Guid:
+                case EdmPrimitiveTypeKind.Int16:
+                case EdmPrimitiveTypeKind.Int32:
+                case EdmPrimitiveTypeKind.Int64:
+                case EdmPrimitiveTypeKind.SByte:
+                case EdmPrimitiveTypeKind.Single:
+                case EdmPrimitiveTypeKind.Stream:
+                case EdmPrimitiveTypeKind.PrimitiveType:
+                    return new EdmPrimitiveTypeReference(type, isNullable);
+                case EdmPrimitiveTypeKind.Binary:
+                    return new EdmBinaryTypeReference(type, isNullable);
+                case EdmPrimitiveTypeKind.String:
+                    return new EdmStringTypeReference(type, isNullable);
+                case EdmPrimitiveTypeKind.Decimal:
+                    return new EdmDecimalTypeReference(type, isNullable);
+                case EdmPrimitiveTypeKind.DateTimeOffset:
+                case EdmPrimitiveTypeKind.Duration:
+                case EdmPrimitiveTypeKind.TimeOfDay:
+                    return new EdmTemporalTypeReference(type, isNullable);
+                case EdmPrimitiveTypeKind.Geography:
+                case EdmPrimitiveTypeKind.GeographyPoint:
+                case EdmPrimitiveTypeKind.GeographyLineString:
+                case EdmPrimitiveTypeKind.GeographyPolygon:
+                case EdmPrimitiveTypeKind.GeographyCollection:
+                case EdmPrimitiveTypeKind.GeographyMultiPolygon:
+                case EdmPrimitiveTypeKind.GeographyMultiLineString:
+                case EdmPrimitiveTypeKind.GeographyMultiPoint:
+                case EdmPrimitiveTypeKind.Geometry:
+                case EdmPrimitiveTypeKind.GeometryPoint:
+                case EdmPrimitiveTypeKind.GeometryLineString:
+                case EdmPrimitiveTypeKind.GeometryPolygon:
+                case EdmPrimitiveTypeKind.GeometryCollection:
+                case EdmPrimitiveTypeKind.GeometryMultiPolygon:
+                case EdmPrimitiveTypeKind.GeometryMultiLineString:
+                case EdmPrimitiveTypeKind.GeometryMultiPoint:
+                    return new EdmSpatialTypeReference(type, isNullable);
+                default:
+                    throw new NotImplementedException();
+            }
+        }*/
+
+
+        /// <summary>
+        /// Attempts multiple casts on Data.Edm.IEdmType to reach derived classes and extract full name.
+        /// Implementation taken (copy-pasted and primitiveType case commented) from OData V4 ExtensionMethods.
+        /// </summary>
+        /// <param name="type">OData V3 type to extract name of</param>
+        /// <returns>Full name of V3 type</returns>
+        public static string FullTypeName(this Data.Edm.IEdmType type)
+        {
+            // Taken from OData v4 EdmConstants (internal)
+            const string CollectionTypeFormat = "Collection" + "({0})";

```

Comment:
Why not just "Collection({0})" instead of concatenating the strings? The compiler will probably do this for you, but not clear why we would want to concat it here.

Reply:
If it's used only once, I think it's fine to use the string literal directly.